docs/datamodel/functions.rst

```diff
@@ -29,6 +29,34 @@ cardinality:
     {2}
 
 
+User-defined Functions
+----------------------
+
+It is also possible to define custom functions. For example, consider
+a function that adds an exclamation mark ``'!'`` at the end of the
+string or produces ``'!!!'`` if the input is omitted or empty:
```

Comment:
The `!!!` part is unnecessary, let's keep it stupid-simple. Just concatenating `!` is sufficient.

Reply:
Fixed.